### PR TITLE
refactor(express-tooling): clean

### LINKED ISSUE #55: clean up express middleware code
* Add typings where possible
* Remove Set usage where not needed
* Try to not use `any` instead of is-a `req: express.Request`

### PR DESCRIPTION
**Pull request by ISO50**
- add types
- minor improvements

```
---------------------|----------|----------|----------|----------|----------------|
File                 |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
---------------------|----------|----------|----------|----------|----------------|
All files            |     97.1 |    92.86 |    94.55 |    97.55 |                |
 src                 |    96.69 |    93.24 |    93.75 |    97.19 |                |
  TokenCache.ts      |    92.68 |      100 |    83.33 |    94.87 |          60,62 |
  constants.ts       |      100 |      100 |      100 |      100 |                |
  express-tooling.ts |    94.74 |    85.71 |      100 |    94.59 |          52,89 |
  oauth-tooling.ts   |    98.18 |    92.86 |      100 |    98.18 |            225 |
  utils.ts           |      100 |    96.77 |      100 |      100 |             27 |
 src/mock-tooling    |      100 |       90 |      100 |      100 |                |
  index.ts           |      100 |       90 |      100 |      100 |             62 |
---------------------|----------|----------|----------|----------|----------------|

=============================== Coverage summary ===============================
Statements   : 97.1% ( 201/207 )
Branches     : 92.86% ( 78/84 )
Functions    : 94.55% ( 52/55 )
Lines        : 97.55% ( 199/204 )
================================================================================
```

close #55

### CODE REVIEW

**Comment by ISO50** [comment]
Just a preview. Let's discuss if we like for example that async / await stuff.

**Comment by Retro64** [comment]
nyc should exclude the whole build dir: "lib/**/*"

File: src/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export * from './Bodyparameters';
```
Comment (Retro64) [inline src/types/index.ts]: src/types/index.ts(1,15): error TS2307: Cannot find module './Bodyparameters'.

Comment (Retro64) [inline src/types/index.ts]: export * from './BodyParameters';

**Comment by ISO50** [comment]
@Retro64 
>nyc should exclude the whole build dir: "lib/**/*"

```
"exclude": [
      "coverage/**/*.js",
      "lib/integration-test/**/*",
      "lib/src/index.js",
      "lib/test/**/*",
      "lib/**/*"
    ],
```


```
----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |  Unknown |  Unknown |  Unknown |  Unknown |                |
----------|----------|----------|----------|----------|----------------|

=============================== Coverage summary ===============================
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
================================================================================
```

**Comment by Retro64** [comment]
Sry - meant lib/src/types to be excluded

**Comment by Retro64** [comment]
:+1:

**Comment by ISO50** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Comment by bzums** [comment]
👍

**Pull request merged**